tests/expr_and_series/when_test.py

```diff
@@ -144,3 +138,63 @@ def test_when_then_otherwise_multi_output(constructor: Constructor) -> None:
         df.select(x1=nw.when(nw.all() > 1).then(nw.col("a", "b")))
     with pytest.raises(MultiOutputExpressionError):
         df.select(x1=nw.when(nw.all() > 1).then(nw.lit(1)).otherwise(nw.all()))
+
+
+@pytest.mark.parametrize(
+    ("condition", "then", "otherwise", "expected"),
+    [
+        (nw.col("a").sum() == 6, 100, 200, [100]),
+        (nw.col("a").sum() == 6, nw.col("a"), 200, [1, 2, 3]),
+        (nw.col("a").sum() == 6, nw.col("a"), nw.col("b"), [1, 2, 3]),
+        (nw.col("a").sum() == 6, 100, nw.col("b"), [100, 100, 100]),
+        (nw.col("a").sum() == 5, 100, 200, [200]),
+        (nw.col("a").sum() == 5, nw.col("a"), 200, [200, 200, 200]),
+        (nw.col("a").sum() == 5, nw.col("a"), nw.col("b"), [4, 5, 6]),
+        (nw.col("a").sum() == 5, 100, nw.col("b"), [4, 5, 6]),
```

Comment:
similarly here, we should remove test cases: 2, 3, 4, 6, 7, 8 (i started counting at 1 😳 )